app/services/validator_check_active_service.rb

```diff
@@ -27,6 +29,16 @@ def update_validator_activity
 
   private
 
+  def should_be_destroyed?(validator)
+    if !validator.validator_histories.where("created_at > ?", @delinquent_time.ago).exists?
+      if validator.validator_histories.where('created_at < ?', @delinquent_time.ago).exists?
```

Comment:
Why nested instead of &&? Also, this type of negation in this case is not very clear, it's easy to miss, I'd use `empty?` instead.